### ISSUE TITLE
extend module metadata to differentiate "stock" vs non-stock modules 

### ISSUE THREAD

**Issue by roycewilliams** [hashcat/hashcat#3252]
I think there is potential usefulness in marking a module as being "stock" (included with hashcat) vs custom/local.

This would allow hashcat or other tools to treat them differently. For example (and to be clear, this isn't a request for these things, just some examples), custom modules could be listed separately or marked differently in the help output, or they could be included or excluded from benchmarks or hash info, etc. These differences would be invisible to the user _until_ a custom module was added, so most users would never notice. And for users/sites that use custom modules heavily, it might be useful to better manage them (and they might have other suggestions about how it would help).

**Comment by jsteube** [comment]
Fantastic, will add.

**Comment by roycewilliams** [comment]
Looks like this was implemented in 4b4f9b61b9bc54d10eeb39c3a38a3cd912acaf8c / 5ef7b737a141b50d24356e121aec4beb1768e4b7 / 6fce6fb3ff120ed16b300af97cf2144b36edcbe8 / c6c816518c937907affd5d252fe995221ca04e9b ?

**Comment by jsteube** [comment]
Yeah, just wondering if you have other Idea how to make use of the new flag.

**Comment by roycewilliams** [comment]
OK, a couple of ideas:

* If custom modules are detected, list them separately in `hashcat --help` output, or list them with an asterisk next to the mode number, or something (your preference)
* Allow some way to only benchmark custom modules, or to exclude custom modules from `--benchmark-all` (not sure about the best way to do this)

**Comment by Chick3nman** [comment]
I believe more changes were made here, do we want to leave this open or close as completed?

**Comment by roycewilliams** [comment]
Ah, what other changes were made?

**Comment by Chick3nman** [comment]
https://github.com/hashcat/hashcat/commit/5ef7b737a141b50d24356e121aec4beb1768e4b7 at least

**Comment by Chick3nman** [comment]
That may have been an original change actually, i read the commit tag wrong.
